Data/ByteString/Builder/RealFloat/TableGenerator.hs

```diff
@@ -161,6 +164,23 @@ splitWord128s li
   = [fromInteger w | x <- li, w <- [x .&. maxWord64, x `shiftR` 64]]
   where  maxWord64 = toInteger (maxBound :: Word64)
 
+splitWord128 :: Integer -> (Word64,Word64)
+splitWord128 x = (fromInteger (x `shiftR` 64), fromInteger (x .&. maxWord64))
+  where  maxWord64 = toInteger (maxBound :: Word64)
+
+
+case64 :: (Int -> Integer) -> [Int] -> String
+case64 f range = concat
+  [ show i ++ " -> 0x" ++ showHex (f i) "\n"
+  | i <- range]
+
+case128 :: (Int -> Integer) -> [Int] -> String
+case128 f range = concat
+  [ show i ++ " -> (0x" ++ showHex hi "" ++ ", 0x" ++ showHex lo ")\n"
+  | i <- range
+  , let (hi,lo) = splitWord128 (f i)
+  ]
```

Comment:
I'm unsure how these will be used but they seem liable to memory leaks. Specifically the call `f i` in `case64` will be a thunk unless `showHex` scrutinizes it. Similarly `hi` and `lo` and `f i` will also be thunks.

Reply:
I've used a comment explaining where they are used. Performance doesn't matter as they have been run once (by me) to generate the static code. I've considered using `showsPrec` and stuff, but why bother, it's fast enough already.